### ISSUE TITLE
Metal Gear Solid 2: Transparent textures are not displayed properly

### ISSUE THREAD

**Issue by thegamingrestoration** [PCSX2/pcsx2#2514]
***PCSX2 version:***
PCSX2 1.5.0-20180527152736 (4ebfcef)

***PCSX2 options:***
‘Safest’

***Plugins used:***
GSDX, Opengl (Also happens using the Direct3D renderer on Windows)

***Description of the issue:***
This seems to be an old issue that was never fixed: MGS2 uses a few transparent textures that the emulator is incapable of rendering, no matter the settings used.
These textures are shown in a few occasions:

- For Otacon’s glasses (you’re unable to see his eyes during the whole game, except in one cutscene when he takes them off).

- The harrier’s cockpit (you’re supposed to see Solidus inside the cockpit, but PCSX2 renders the crystal opaque).

- The water surface at the very beginning of the Plant chapter, or at the Shell 1 Core (PCSX2 renders the water opaque).

Strangely enough, it only seems to affect these cases, the rest of the transparent textures are displayed properly.

- **GS dump of Otacon with his glasses on**: 
[gsdx_20180718192951.gs.zip](https://github.com/PCSX2/pcsx2/files/2206790/gsdx_20180718192951.gs.zip)

***PC specifications:***
CPU: Intel i7 2600
GPU: GT 1030
OS: Linux 64 Bit

***Screenshots showing a few examples:***
![gsdx_20180718192951](https://user-images.githubusercontent.com/37403401/42898503-e29c5d60-8ab2-11e8-8361-52b158251522.png)
![gsdx_20180718193227](https://user-images.githubusercontent.com/37403401/42898504-e2b392fa-8ab2-11e8-8dac-202e88fa6949.png)
![gsdx_20180718193513](https://user-images.githubusercontent.com/37403401/42898505-e2d4c4fc-8ab2-11e8-9654-896227c9d6bf.png)

**Comment by RhythmLady** [comment]
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ
https://discord.gg/bFCfZte
https://discord.gg/N24gCZJ

**Comment by atomic83GitHub** [comment]
So that problem also affect MGS3 too:

How it is on PCSX2:

![image](https://user-images.githubusercontent.com/26351275/62961345-e1403300-bdfc-11e9-8d82-50028e1c2f23.png)

How it should be:

![image](https://user-images.githubusercontent.com/26351275/62961383-ed2bf500-bdfc-11e9-9caf-c070905d4136.png)

**Comment by tadanokojin** [comment]
Please try sw mode in #3444

**Comment by v-fox** [comment]
Why was this closed ? I just tested 1.7.0~dev+233~git20200912T145909 x86_64 build and in the intro codec conversations Otacon’s glasses are opaque even with maxed out blending.

**Comment by refractionpcsx2** [comment]
Please try tadanokojin's suggestion with a 32bit build

**Comment by v-fox** [comment]
> Please try tadanokojin's suggestion with a 32bit build

You mean unmerged PR, horrible-looking SW renderer and out-of-place 32bit build that requires duplicating entire core part of OS to run ? Then I might as well run Windows version under wine with DX output or MGS HD Collection in RPCS3.

Even if it fixes it, it still unmerged, sw-only and 32bit-only. Not something that could be considered fixed, so why is this closed ?

**Comment by tadanokojin** [comment]
Because the author of the issue closed it? Ask him.

**Comment by v-fox** [comment]
> Because the author of the issue closed it? Ask him.

Which is what I'm doing. There are these things called "notifications", usually enabled by default for thread authors, very cool.
But aren't you have rights to forcibly close and open those ?

**Comment by lightningterror** [comment]
```
Which is what I'm doing. There are these things called "notifications", usually enabled by default for thread authors, very cool.
But aren't you have rights to forcibly close and open those ?
```
You do realize that you can ask why a thread was closed and request for it to be re open in a civil manner without sounding like a total jackass right?

**Comment by v-fox** [comment]
@lightningterror 
> ```
> Which is what I'm doing. There are these things called "notifications", usually enabled by default for thread authors, very cool.
> But aren't you have rights to forcibly close and open those ?
> ```
> 
> You do realize that you can ask why a thread was closed and request for it to be re open in a civil manner without sounding like a total jackass right?

Do you realize you can answer a simple question without sounding like a total jackass or do anything useful other than coming just for giving lip due to some personal psychological problems, right ? Especially when you're a genius that comes up with stuff like https://github.com/PCSX2/pcsx2/pull/2788 that then has to be https://github.com/PCSX2/pcsx2/commit/a1bf6775144a073746e0b37d59a5182279511aea

You know what else one can do ? Figure out that glitches don't just fix themselves and that maybe the most hardware intensive games, a benchmark of its time, and one of the reasons for universal interest in an emulator warrants at least passing attention.

@refractionpcsx2 
Anyway, I patched the build with https://github.com/PCSX2/pcsx2/pull/3444 while working around https://github.com/PCSX2/pcsx2/issues/3691 and, indeed, glasses are fine in SW-mode. Ironically, original PC release of MGS2 also had these transparency issues which was then patched. So, devs had such troubles while porting it even back then.

**Comment by Terminento** [comment]
I'm totally profane to cding or compiling, how can I patch the build with #3444 ? Is there a compiled plugin that already contains the fix? Thanks in advance.

**Comment by Helium-4** [comment]
You can grab a compiled build here
https://ci.appveyor.com/project/gregory38/pcsx2/builds/33359184/job/91s6yy59kkocb8k6/artifacts

**Comment by Terminento** [comment]
Thanks a lot! It works like a charm in software mode, so now there's a way to play the game in its original form. 
I feel curious, is there a reason for this fix no to be implemented in the main build? Maybe it breaks other effects in other games?

**Comment by refractionpcsx2** [comment]
Some of the GSdx guys are not completely happy with the current implementation, also hardware mode needs to be done, so there's more work to be completed before we can merge

**Comment by Terminento** [comment]
I see. Looking forward those issues being fixed in teh future! Keep up the good work, guys!

**Comment by suyash691** [comment]
Hey unfortunately the build has exceeded its retention policy. Can anyone share the build?

**Comment by Sergeanur** [comment]
I fixed this in #4385
